fix(oxfmt): Do not follow symlinks like prettier

leaysgur opened the pull request:
Part of #10573

> Prettier no longer follows symbolic links while expanding command line arguments. This avoids problems in many scenarios such as symlinks outside the source tree, symlinks to ignored files, and cycles of symlinks.
> https://prettier.io/blog/2023/07/05/3.0.0.html#dont-expand-globs-via-symbolic-links-14627-by-andersk

Follows this behavior.

leaysgur commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13606?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#13606** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13606?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13606?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#13605** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13605?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13606).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen reviewed (changes requested):
Can you inline the reference comment and link? We always reference with URLs inside comments to provide maximum context in the code.

leaysgur commented:
Thank you~. Updated: https://github.com/oxc-project/oxc/pull/13606/commits/9442364dc285c96266753f039dfb9339eb3addfc

leaysgur committed:
fix(oxfmt): Do not follow symlinks like prettier (#13606)

Part of #10573

> Prettier no longer follows symbolic links while expanding command line arguments. This avoids problems in many scenarios such as symlinks outside the source tree, symlinks to ignored files, and cycles of symlinks.
> https://prettier.io/blog/2023/07/05/3.0.0.html#dont-expand-globs-via-symbolic-links-14627-by-andersk

Follows this behavior.
